### DISCUSSION TITLE
(SOLVED) rejectOnNotFound warning message when i call findMany()

### DISCUSSION THREAD

**Discussion by darkodavidovic** [prisma/prisma#14858] [Q&A] [answered]
My query:
export const users = queryField("Users", {
  type: nonNull(list(nonNull(User))),
  resolve: async (_root, _args, ctx) => {
    return ctx.prisma.user.findMany()
  }
});
Error i see everythime on this query in node terminal: 
`rejectOnNotFound` option is deprecated and will be removed in Prisma 5. Please use `prisma.user.findFirstOrThrow` method instead

**Comment by austincrim** [comment, marked as answer]
Hey @darkodavidovic 👋,

When you create the `PrismaClient` instance, are you using the `rejectOnNotFound` option anywhere?

**Reply by darkodavidovic** [reply]
Nope. Inside .prisma file in generator client are just provider = "prisma-client-js". The rest of the data is just basic User model for my table. Prisma version is 4.2.1

**Reply by austincrim** [reply]
Can you share the code snippet where you call `new PrismaClient`?

**Reply by darkodavidovic** [reply]
import { PrismaClient } from "@prisma/client";
export const prisma = new PrismaClient();

**Reply by austincrim** [reply]
Thanks!

Are you certain the above query is the one triggering the console warning? Are you calling a `findFirst` anywhere in your code?

**Reply by austincrim** [reply]
Looks like you're using `rejectOnNotFound` [here in your code](https://github.com/darkodavidovic/prisma-nexus-v1/blob/main/src/context.ts#L27-L34).

As the warning says, this option will be deprecated in the future and should be replaced with `findFirstOrThrow.

**Reply by darkodavidovic** [reply]
Ah! Warning is gone after I got rid of findFirst and findUnique from all my queries.
- findFirst renamed to findFirstOrThrow
- findUnique renamed to findUniqueOrThrow
- Removed "rejectOnNotFound: true" bellow where conditions.
The Warnig started after i updated to latest prisma. Old code got stuck using old deprecated methods.
Thank you for hints.  :-)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
